components/Chart/index.tsx

```diff
@@ -0,0 +1,260 @@
+        prices.map((x) => x.timestamp),
+        x0,
+        1
+      )
+
+      const d0 = prices[index - 1]
+      const d1 = prices[index]
+      let pricePoint = d0
+
+      const hasPreviousData = d1 && d1.timestamp
+      if (hasPreviousData) {
+        pricePoint = x0.valueOf() - d0.timestamp.valueOf() > d1.timestamp.valueOf() - x0.valueOf() ? d1 : d0
+      }
+
+      if (pricePoint) {
+        setCrosshair(timeScale(pricePoint.timestamp))
+        setDisplayPrice(pricePoint)
+      }
+    },
+    [timeScale, prices]
+  )
+
+  const resetDisplay = useCallback(() => {
+    setCrosshair(null)
+    setDisplayPrice(endingPrice)
+  }, [setCrosshair, setDisplayPrice, endingPrice])
+
+  const mainColor = getPriceChangeColor(priceChange)
+
+  return (
+    <svg data-cy="price-chart" width={width} height={height} style={{ minWidth: '100%', maxWidth: '100%' }}>
+      <LineChart
+        width={width}
+        height={height}
+        data={prices}
+        getX={getX}
+        getY={getY}
+        marginTop={margin.top}
+        curve={curve}
+        strokeWidth={2}
+        color={mainColor}
+      />
+
+      {crosshair !== null ? (
+        <g>
+          <AxisBottom
+            scale={timeScale}
+            stroke={Color.text1}
+            tickFormat={tickFormatter}
+            tickStroke={Color.text1}
+            tickLength={4}
+            hideTicks={true}
+            tickTransform="translate(0 -5)"
+            tickValues={updatedTicks}
+            top={height - 1}
+            tickLabelProps={() => ({
+              fill: Color.text1,
+              fontSize: 11,
+              textAnchor: 'middle',
+              transform: 'translate(0 -24)',
+            })}
+          />
+          <text
+            x={crosshair + (crosshairAtEdge ? -4 : 4)}
+            y={margin.crosshair + 10}
+            textAnchor={crosshairAtEdge ? 'end' : 'start'}
+            fontSize={15}
+            fill={Color.text1}
+          >
+            {crosshairDateFormatter(displayPrice.timestamp)}
+          </text>
+          <Line
+            from={{ x: crosshair, y: margin.crosshair }}
+            to={{ x: crosshair, y: height }}
+            stroke={Color.text1}
+            strokeWidth={1}
+            pointerEvents="none"
+            strokeDasharray="4,4"
+          />
+          <GlyphCircle
+            left={crosshair}
+            top={rdScale(displayPrice.value) + margin.top}
+            size={50}
+            fill={mainColor}
+            stroke={mainColor}
+            strokeWidth={0.5}
+          />
+        </g>
+      ) : (
+        <AxisBottom hideAxisLine={true} scale={timeScale} stroke={Color.text1} top={height - 1} hideTicks />
+      )}
+      <rect
+        x={0}
+        y={0}
+        width={width}
+        height={height}
+        fill="transparent"
+        onTouchStart={handleHover}
+        onTouchMove={handleHover}
+        onMouseMove={handleHover}
+        onMouseLeave={resetDisplay}
+      />
+    </svg>
+  )
+}
```

Comment:
I guess this is mostly C&P from Uniswap widget, if it's the case maybe we can leave until we can clean up a bit, but if we could it would be good to simplify this component. The render function is too big